support/k8s/k8s-workload-registrar/config_crd.go

```diff
@@ -51,6 +55,19 @@ func (c *CRDMode) ParseConfig(hclConfig string) error {
 		c.WebhookPort = defaultWebhookPort
 	}
 
+	if c.IdentityTemplate != "" && (c.PodAnnotation != "" || c.PodLabel != "") {
+		return errs.New("workload registration configuration is incorrect, can only use one of identity_template, pod_annotation, or pod_label")
+	}
+
+	// If no identity format set, use the default identity template.
+	if c.Mode == modeCRD && c.IdentityTemplate == "" && c.PodAnnotation == "" && c.PodLabel == "" {
```

Comment:
i don't think you need `c.Mode == modeCRD` because we won't get to this point if it was some other mode configured.